tools/analyzer_plugin/tool/doc_generation_utils/generate_contributor_doc_page.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:io';
+
+import 'package:markdown/markdown.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/documented_contributor_meta.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/maturity.dart';
+import 'package:over_react_analyzer_plugin/src/util/constants.dart';
+
+import 'shared_html.dart';
+
+abstract class ContributorDocPage<T extends DocumentedContributorMetaBase> {
+  final T contributor;
+  ContributorDocPage(this.contributor);
+
+  String get maturityString {
+    switch (contributor.maturity) {
+      case Maturity.deprecated:
+        return '<span style="color:orangered;font-weight:bold;" >${contributor.maturity}</span>';
+      case Maturity.experimental:
+        return '<span style="color:hotpink;font-weight:bold;" >${contributor.maturity}</span>';
+      default:
+        return '${contributor.maturity}';
+    }
+  }
+
+  void generate(String filePath) {
+    var generated = _generate();
+    if (filePath != null) {
+      var outPath = '$filePath/${contributor.name}.html';
+      print('Writing to $outPath');
+      File(outPath).writeAsStringSync(generated);
```

Comment:
This fails if the parent dir doesn't exist
```suggestion
      File(outPath)
        ..parent.createSync(recursive: true)
        ..writeAsStringSync(generated);
```